pkg/base/helm.go

```diff
@@ -83,12 +85,39 @@ func RenderHelm(u *upstreamtypes.Upstream, renderOptions *RenderOptions) (*Base,
 
 	baseFiles := []BaseFile{}
 	for k, v := range rendered {
-		baseFile := BaseFile{
-			Path:    k,
-			Content: []byte(v),
+		if !renderOptions.SplitMultiDocYAML {
+			baseFile := BaseFile{
+				Path:    k,
+				Content: []byte(v),
+			}
+
+			baseFiles = append(baseFiles, baseFile)
+			continue
+		}
+
+		fileStrings := strings.Split(v, "\n---\n")
+		if len(fileStrings) == 1 {
+			baseFile := BaseFile{
+				Path:    k,
+				Content: []byte(v),
+			}
+
+			baseFiles = append(baseFiles, baseFile)
+			continue
 		}
 
-		baseFiles = append(baseFiles, baseFile)
+		for idx, fileString := range fileStrings {
+			filename := strings.TrimSuffix(k, filepath.Ext(k))
+			filename = fmt.Sprintf("%s-%d%s", filename, idx, filepath.Ext(k))
+
+			baseFile := BaseFile{
+				Path:    filename,
+				Content: []byte(fileString),
+			}
+
+			baseFiles = append(baseFiles, baseFile)
+			continue
```

Comment:
this `continue` is confusing IMO
since it's at the end of a for loop and is thus redundant